cpp/src/arrow/util/compression.h

```diff
@@ -107,6 +108,40 @@ class ARROW_EXPORT Decompressor {
   // XXX add methods for buffer size heuristics?
 };
 
+/// \brief Compression codec options
+class ARROW_EXPORT CodecOptions {
+ public:
+  explicit CodecOptions(int compression_level = kUseDefaultCompressionLevel) {
+    compression_level_ = compression_level;
+  }
+  virtual ~CodecOptions() = default;
+
+  int compression_level_ = kUseDefaultCompressionLevel;
```

Comment:
Forgot this, but style nit (this is a public member, it's weird to add a trailing underscore):
```suggestion
  int compression_level = kUseDefaultCompressionLevel;
```